app/javascript/controllers/home_search_controller.js

```diff
@@ -2,8 +2,118 @@ import { Controller } from "@hotwired/stimulus"
 
 // Connects to data-controller="home-search"
 export default class extends Controller {
+  static targets = [ "input", "dropDown", "ontology", "searchedOntologies", "searchOntologyContent", "homeSearchOntologies" ]
+  static values = {
+    ontologies: Array
+  }
   connect() {
+    this.ontologies = this.ontologiesValue
+    this.input = this.inputTarget
+    this.dropDown = this.dropDownTarget
+    this.ontology = this.ontologyTarget
+    this.searchedOntologies = this.searchedOntologiesTarget
+    this.searchOntologyContent = this.searchOntologyContentTarget
+    this.homeSearchOntologies = this.homeSearchOntologiesTarget
+    
+    this.input.addEventListener("input",()=>{
+      this.#searchInput()
+    } );
+    this.dropDown.addEventListener("mousedown", function(event) {
+        event.preventDefault(); // prevent the blur event from firing
+    });
+
+    this.input.addEventListener("blur", function() {
+        
+        this.dropDown.style.display = "none";
+        this.input.classList.remove("home-dropdown-active");
+    });
+
   }
 
   
+  #scrollDown(currentScroll) {
+    const startPosition = window.pageYOffset;
+    const distance = 300 - currentScroll;
+    const duration = 1000;
+    let start = null;
+
+    function scrollAnimation(timestamp) {
+      if (!start) start = timestamp;
+      const progress = timestamp - start;
+      const scrollPosition = startPosition + easeInOutCubic(progress, 0, distance, duration);
+      window.scrollTo(0, scrollPosition);
+      if (progress < duration) {
+        window.requestAnimationFrame(scrollAnimation);
+      }
+    }
+
+    function easeInOutCubic(t, b, c, d) {
+      t /= d / 2;
+      if (t < 1) return c / 2 * t * t * t + b;
+      t -= 2;
+      return c / 2 * (t * t * t + 2) + b;
+    }
+
+    window.requestAnimationFrame(scrollAnimation);
+  }
+
+  #searchInput() {
+    let results_list = []
+    const inputValue = this.input.value.trim();
+    if (inputValue.length > 0) {
+        this.ontology.innerHTML = inputValue;
+        this.searchedOntologies.innerHTML = inputValue;
+        this.searchOntologyContent.href = "/search?query="+inputValue;
```

Comment:
extract this to a constant at the top `"/search?query="` and `"/ontologies?search="`